Data-Structures/Linked-List/SinglyLinkedList.js

```diff
@@ -21,112 +21,116 @@ class LinkedList {
     this.length = 0
   }
 
+  initiateNodeandIndex () {
+    return { currentNode: this.headNode, currentIndex: 0 }
+  }
+
   // Returns length
   size () {
     return this.length
   }
 
   // Returns the head
   head () {
-    return this.headNode ? this.headNode.data : null
+    return this.headNode?.data || null
+  }
+
+  // Return if the list is empty
+  isEmpty () {
+    return this.length === 0
   }
 
   // add a node at last it to linklist
   addLast (element) {
-    const node = new Node(element)
-
     // Check if its the first element
     if (this.headNode === null) {
-      this.headNode = node
-    } else {
-      let currentNode = this.headNode
-
-      // Loop till there is a node present in the list
-      while (currentNode.next) {
-        currentNode = currentNode.next
-      }
+      return this.addFirst(element)
```

Comment:
You are not increasing the length anymore.

Reply:
The add first method will increase the length so it is not needed here.
``` javascript
addFirst (element) {
    const node = new Node(element)
    node.next = this.headNode
    this.headNode = node
    this.length++
    return this.size()
  }
```